pkg/logcli/client/client.go

```diff
@@ -216,21 +217,39 @@ func (c *DefaultClient) doRequest(path, query string, quiet bool, out interface{
 	if c.Tripperware != nil {
 		client.Transport = c.Tripperware(client.Transport)
 	}
-	resp, err := client.Do(req)
-	if err != nil {
-		return err
+
+	var resp *http.Response
+	retries := c.Retries
+	success := false
+
+	for retries > 0 {
```

Comment:
If a user sets the retries flag value to 0 by mistake then they would unexpectedly skip making any calls. Also, retried calls should be the ones done after seeing a failure in the initial call, so let us initialise the `retries` above to `c.Retries + 1`.
What do you think?

Reply:
You are right, but I enhanced your idea a little bit. I renamed the variable to "attempts", which seemed like a better name to me now. I implemented your suggestion, so `attempts = c.Retries + 1` and I made the default value for c.Retries = 0. So there still is 1 attempt unless the user sets the retries flag to something other than 0.